sqlglot/parser.py

```diff
@@ -2049,7 +2049,23 @@ def _parse_function(self, functions=None):
             args = self._parse_csv(self._parse_lambda)
 
             if function:
-                this = function(args)
+                params = None
+
+                # Clickhouse supports function calls like foo(x, y)(z), so for these we need to also parse the
+                # second parameter list (i.e. "(z)") and the corresponding function will receive both arg lists.
+                if function.__code__.co_name == "<lambda>" and function.__code__.co_argcount == 2:
```

Comment:
The `<lambda>` check was added because without it we would also match functions like `format_time_lambda` which expects two arguments (but the second one is a helper we hard code into the calls, if I'm not mistaken).

Reply:
this doesn't seem correct to me, format_time_lambda should return a lambda that only has one argument

Reply:
fixed.